src/bandersnatch_filter_plugins/allowlist_name.py

```diff
@@ -202,6 +202,17 @@ def _determine_filtered_package_requirements(self) -> list[Requirement]:
             package_lines = []
         return list(_parse_package_lines(package_lines))
 
+    def pinned_version_exists(self, metadata: dict) -> bool:
+        name = canonicalize_name(metadata["info"]["name"])
+        for requirement in self.allowlist_release_requirements:
+            if name != requirement.name:
+                continue
+            elif len(requirement.specifier) > 0:
+                return True
+            else:
+                return False
```

Comment:
```suggestion
            return len(requirement.specifier) > 0
```
Just a NIT for less lines